config/config.go

```diff
@@ -147,6 +147,9 @@ type Config struct {
 
 	// HTTP request interceptor, that assigns Authorization header
 	auth func(r *http.Request) error
+
+	// Keep track of the source of each attribute
+	attrSource map[string]*Source
```

Comment:
The `*Source` elements of this map are never mutated, so they can (should?) be a regular value, not a ptr.